runtime/io/common/src/aws/smithy/kotlin/runtime/io/SdkByteChannel.kt

```diff
@@ -20,23 +21,26 @@ import io.ktor.utils.io.ByteReadChannel as KtorByteReadChannel
  */
 public interface SdkByteChannel : SdkByteReadChannel, SdkByteWriteChannel {
     override fun close() {
-        (this as SdkByteWriteChannel).close()
+        close(null)
     }
 }
 
 /**
  * Create a buffered channel for asynchronous reading and writing of bytes
+ * @param autoFlush Flag indicating if the channel should auto flush after every write, see [SdkByteWriteChannel.autoFlush]
  */
 @InternalApi
-public fun SdkByteChannel(autoFlush: Boolean = true): SdkByteChannel =
-    KtorByteChannel(autoFlush).toSdkChannel()
+public fun SdkByteChannel(
+    autoFlush: Boolean = true,
+    maxBufferSize: Int = DEFAULT_BYTE_CHANNEL_MAX_BUFFER_SIZE,
+): SdkByteChannel = RealSdkByteChannel(autoFlush, maxBufferSize)
 
 /**
- * Creates a channel for reading from the given byte array.
+ * Creates a channel for reading with the contents of the given byte array.
  */
 @InternalApi
 public fun SdkByteReadChannel(
     content: ByteArray,
     offset: Int = 0,
     length: Int = content.size - offset,
-): SdkByteReadChannel = KtorByteReadChannel(content, offset, length).toSdkChannel()
+): SdkByteReadChannel = RealSdkByteChannel(content, offset, length).apply { close() }
```

Comment:
**Question**: Would closing the channel be better done in the `ByteArray`-accepting constructor? Or do we anticipate a future use case for creating a read channel with an initial byte array and then adding more later?

Reply:
I'm not sure why I put it here, I see no reason it can't be moved to the constructor at first glance though